packages/api/internal/handlers/template_start_build.go

```diff
@@ -88,93 +90,88 @@ func (a *APIStore) PostTemplatesTemplateIDBuildsBuildID(c *gin.Context, template
 		attribute.String("template.id", templateID),
 	)
 
-	// Create a new build cache for storing logs
-	err = a.buildCache.Create(templateID, buildUUID, team.ID)
-	if err != nil {
-		a.sendAPIStoreError(c, http.StatusConflict, fmt.Sprintf("there's already running build for %s", templateID))
+	// Check if some other build from same template is in process, excludes currently created build
+	envBuildDB, err := a.db.GetRunningEnvBuild(ctx, envDB.ID, []uuid.UUID{buildUUID})
+
+	// error is present but different from not found
+	if err != nil && !errors.Is(err, db.TemplateBuildNotFound{}) {
```

Comment:
This is hard to read and I think you even confused yourself, the second if branch is only satisfied if
```
errors.Is(err, db.TemplateBuildNotFound{})
```
and the comment is `we found conflicting build for same env`

Reply:
yea, i changed it and make it more readable